Added InventorySwapEvent - BUKKIT-3386

seifpic opened the pull request:
The InventorySwapEvent is fired when a Player swaps items to/fro the hotbar to/fro the inventory using the numberkeys. This was a feature added in Minecraft Snapshot 12w40a. In previous bukkit versions, there was no way to detect this, however I noticed that the numberKeys were sent through protocol 102 in the MouseButton variable. Also HoldingShift equals 2 when numberKeys are pressed.

InventorySwapEvent extends InventoryClickEvent because it uses mostly the same methods. getHotbarRawSlot() returns the raw slot number of the slot which was swapped to. getHotbarSlot() returns the hotbar slot which is being swapped with the item under the cursor. (from 1-9).

Fixes: https://bukkit.atlassian.net/browse/BUKKIT-3386
CraftBukkit PR: https://github.com/Bukkit/CraftBukkit/pull/1018
Test Plugin: http://pastebin.com/nrfwiAPD

Signed-off-by: seifpic seifpic@hotmail.com

mbax commented on src/main/java/org/bukkit/event/inventory/CraftItemEvent.java:
```diff
@@ -8,8 +8,8 @@
 
-    public CraftItemEvent(Recipe recipe, InventoryView what, SlotType type, int slot, boolean right, boolean shift) {
-        super(what, type, slot, right, shift);
+    public CraftItemEvent(Recipe recipe, InventoryView what, SlotType type, int slot, boolean right, boolean shift, int NumberKey) {
```
Should deprecate previous constructor not overwrite

seifpic replied:
It adds to the constructor, so it won't affect any older plugins, so I though deprecating it would be unnecessary.

mbax replied:
Adding to the constructor would BREAK any older plugins calling the constructor.

mbax commented on src/main/java/org/bukkit/event/inventory/CraftItemEvent.java:
```diff
@@ -8,8 +8,8 @@
-    public CraftItemEvent(Recipe recipe, InventoryView what, SlotType type, int slot, boolean right, boolean shift) {
-        super(what, type, slot, right, shift);
+    public CraftItemEvent(Recipe recipe, InventoryView what, SlotType type, int slot, boolean right, boolean shift, int NumberKey) {
+        super(what, type, slot, right, shift, NumberKey);
```
Uppercase variable here.

mbax commented on src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java:
```diff
@@ -16,8 +16,9 @@
     private int whichSlot;
     private int rawSlot;
     private ItemStack current = null;
+    private int NumberKey;
```
Uppercase variable here

mbax commented on src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java:
```diff
@@ -16,8 +16,9 @@
+    private int NumberKey;
 
-    public InventoryClickEvent(InventoryView what, SlotType type, int slot, boolean right, boolean shift) {
+    public InventoryClickEvent(InventoryView what, SlotType type, int slot, boolean right, boolean shift, int NumberKey) {
```
Should deprecate previous constructor not overwrite

jrtc27 commented:
Fix the indentation!

seifpic commented:
Rewriting tomorrow.

riking commented:
You know, you can rewrite it, then rebase and push -f to your branch and it'll auto-update the PR

seifpic commented:
That's why I didnt delete the branch ;)

feildmaster commented:
But you closed the PR,when you could have used this same one

seifpic committed:
Added InventorySwapEvent - BUKKIT-3386

The InventorySwapEvent is fired when a Player swaps items to/fro the hotbar to/fro the inventory using the numberkeys. This was a feature added in Minecraft Snapshot 12w40a. In previous bukkit versions, there was no way to detect this, however I noticed that the numberKeys were sent through protocol 102 in the MouseButton variable. Also HoldingShift equals 2 when numberKeys are pressed.

InventorySwapEvent extends InventoryClickEvent because it uses mostly the same methods. getHotbarRawSlot() returns the raw slot number of the slot which was swapped to. getHotbarSlot() returns the hotbar slot which is being swapped with the item under the cursor. (from 1-9).

Fixes: https://bukkit.atlassian.net/browse/BUKKIT-3386
CraftBukkit PR:
Test Plugin: http://pastebin.com/nrfwiAPD

Signed-off-by: seifpic <seifpic@hotmail.com>

seifpic commented:
Reopen? ;)
